### PR TITLE
feat(o365): add_new_field_to_parse

### PR DESCRIPTION
**Pull request by tfanouillere-sekoia**
## Description
https://github.com/SekoiaLab/integration/issues/1533

## Type of Change

- [ ] New intake format
- [x] Update to existing intake format
- [ ] Bug fix
- [ ] Documentation update
- [ ] Other (please describe):

## Affected Intake Formats

 Office 365/o365

- Vendor/product:

### 🔒 CRITICAL - Security Requirements (MUST BE VERIFIED FIRST)

- [x] **No real email addresses** in test files (use: user@example.com, test@example.org)
- [x] **No real passwords or API keys** (use: "P@ssw0rd!", "xxxxxxxxxxxx", "FAKE_TOKEN_123")
- [x] **No real IP addresses** (use TEST-NET ranges: 192.0.2.x, 198.51.100.x, 203.0.113.x)
- [x] **No real phone numbers** (use: +1-555-0100 or similar test numbers)
- [x] **No real names or PII** (use: "John Doe", "Jane Smith", "User123")
- [x] **No real company/organization names** in test data (unless public vendors)
- [x] **No real usernames, hostnames, or domain names** (use: user1, host.example.com, test.corp)
- [x] All test data has been anonymized and verified

### Required for All PRs

- [x] Code has been linted
    - [x] with the linter for manifest, smart-descriptions, and test files (`uv run python linter.py check --changes` in utils/)
    - [x] with Prettier for the parser code files (`npx prettier --write <module>/<format>/ingest/parser.yml`)
- [x] All CI/CD checks pass
- [x] Changes follow the contribution guidelines in `CONTRIBUTING.md`

### Required for New/Updated Intake Formats

- [x] Clear descriptions provided for modules and formats
- [ ] Logo files included for new modules/formats
- [x] Parser test coverage is at least 75%
- [ ] At least one smart-description is provided
- [ ] README.md updated (if applicable)
- [x] Taxonomy mappings are correct and documented
- [x] Test cases cover edge cases and error handling

### Testing

- [x] Local tests pass (`uv run pytest` in utils/)
- [x] Sample data is representative and **completely anonymized**
- [x] **No sensitive information** (PII, credentials, real IPs, real emails) in test files
- [x] Test data uses example.com/example.org domains and TEST-NET IP ranges
- [x] Parser handles malformed data gracefully

### Documentation

- [x] Code changes are documented
- [ ] Smart-descriptions are clear and informative
- [x] Field mappings are explained

## Related Issues

Closes #

## Summary by Sourcery

Add support for capturing SharePoint search query text in the Office 365 intake format.

New Features:
- Introduce the office365.audit.search_query.text field to store the text of executed search queries.

Tests:
- Add a SharePoint search query test fixture to validate parsing of the new search query text field.

### CODE REVIEW

**Review by mchupeau-sk — DISMISSED** [review]
LGFM

File: Office 365/o365/_meta/fields.yml [hunk 1]
```diff
@@ -117,6 +117,11 @@ office365.audit.platform:
   name: office365.audit.platform
   type: keyword
 
+office365.audit.search_query.text:
+  description: The text of the search query that was performed
+  name: office365.audit.search_query.text
+  type: keyword
```
Comment (squioc) [inline Office 365/o365/_meta/fields.yml]: I suggest stripping the `.text` from the name of the field:


```suggestion
office365.audit.search_query:
  description: The text of the search query that was performed
  name: office365.audit.search_query
  type: keyword
```

File: Office 365/o365/ingest/parser.yml [hunk 2]
```diff
@@ -553,6 +545,7 @@ stages:
           office365.audit.event_source: "{{json_event.message.EventSource}}"
           office365.audit.item_type: "{{json_event.message.ItemType}}"
           office365.audit.platform: "{{json_event.message.Platform}}"
+          office365.audit.search_query.text: "{{json_event.message.SearchQueryText}}"
```
Comment (squioc) [inline Office 365/o365/ingest/parser.yml]: Related to the previous comment:
```suggestion
          office365.audit.search_query: "{{json_event.message.SearchQueryText}}"
```

**Review by squioc — COMMENTED** [review]
Thank you for the work.

Some changes were suggested

**Comment by mchupeau-sk** [comment]
Duplicated PR
